### ISSUE TITLE
Typescript for Node for Max

### ISSUE THREAD

**Issue by geofholbrook** [Cycling74/n4m-examples#26]
#### Nature of issue?

- [ ] Found a bug in an example
- [ ] Existing example enhancement
- [x] Suggest a new example

#### New example details:

This may not fit in here ... it's a more general request about Typescript! I wonder if anyone has taken a crack at a declaration file for Node for Max? Ideally it would get published to the DefinitelyTyped repository, as @types/node-for-max. Then there could be another category of examples in here, that show how to write .ts files.

**Comment by fde31** [comment]
I 100% agree. I've filed an issue internally to offer types for Node For Max and I assume we'll then highlight that in the core examples.

**Comment by FrancescElies** [comment]
Any updates on this?

We couldn't avoid noticing that there are types inplace `Cycling '74/Max 8/resources/packages/Node for Max/source/@types\max-api.d.ts` but it would be great to have some guidance on how are they intended to be used.

**Update** 04-04-2023: I am afraid `max-api.d.ts` in their current form in Max `8.3` can't be used, but maybe I didn't manage the typescript compiler to swallow them.

**Comment by fde31** [comment]
Hi @FrancescElies - I've submitted a PR with updated / fixed type definitions here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65005

Once that is merged you should be able to install `@types/max-api`. There is an existing version on NPM that @geofholbrook worked on but after testing it locally I've identified a few mismatches in the parameter signatures and the way the API is exported that made it a bit troublesome to use. In the meantime I'd recommend to just grab the [type definitions from the Pull Request](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c9a37155e92605fcc0231fe26c49d0f536cb5d2c/types/max-api/index.d.ts) and using them in your local project:

```typescript
declare module "max-api" {
  // Content of the file on DefinitelyTyped
  // Just make sure to drop the "declare" in front of "namespace MaxAPIStatic"
}
```

I also have a simple, local example that I'll make available once the updated definitions have been published.

**Comment by FrancescElies** [comment]
@fde31 thanks! we are already on the way to integrate them on our side and as soon as the pr is merged we will switcht to the npm package with `npm i @types/max-api -D`

**Comment by geofholbrook** [comment]
@fde31 Great to see @types/max-api updated! I approved the PR as "owner" but you should take primary ownership obviously.

**Comment by fde31** [comment]
`@types/max-api` v2.0.0 is published on NPM and PR #39 adds an example that demonstrates the basics of TS usage in https://github.com/Cycling74/n4m-examples/tree/master/typescript
